src/ScriptEditor/NetscriptDefinitions.d.ts

```diff
@@ -6422,7 +6422,7 @@ export interface NS {
    * @param host - Hostname of target server.
    * @returns The number of open ports required to successfully run NUKE.exe on the specified server.
```

Comment:
This should mention the circumstances where it returns undefined. Other than that, LGTM

Reply:
This did cross my mind. Is there a way to make it clear in a non-spoilery way? This is a function new players are likely to come across.

Reply:
"Returns undefined for servers that don't have openable ports"?

It's a little ambiguous, because arguably the same applies to personal servers and home.

Reply:
If it should return undefined for all player-owned servers, then using the "failOnHacknetServer" function is not great.

Really it's the internal structure that kind of sucks, where BaseServer doesn't have those properties. `Server` is basically a "hackable" server, but home and regular purchased servers are also this type of server even though they're not hackable. Hacknet servers inherit from BaseServer instead of Server, so they end up missing a bunch of properties specific to hackable servers.

Not sure what the best solution is here.

Reply:
Fair points. 

Is it worth putting this on the back-burner, in the event a more thorough rework of these systems could be planned? 

I doubt it would be much work to make things consistent but I suspect those would be breaking changes.

Reply:
IMO it's worth making this more consistent, even if it's not perfect. The situation with hashnet vs pservers vs home extends to much more than just open ports, and it's definitely a larger issue.